fdbserver/team/Serializer.h

```diff
@@ -0,0 +1,110 @@
+#ifndef FDBSERVER_TEAM_ENCODER_H
+#define FDBSERVER_TEAM_ENCODER_H
+
+#pragma once
+
+#include <cstdint>
+#include <cstring>
+#include <vector>
+
+#include "flow/crc32c.h"
+#include "flow/Error.h"
+#include "flow/error_definitions.h"
+#include "flow/serialize.h"
+
+/**
+ * Get the number of bytes of a serialized object. Only works with objects with POD variables.
+ */
+template <typename T>
+size_t getSerializedBytes() {
+	// Due to padding, the serialized object may take less space, have to figure
+	// out by evaluating.
+	static size_t value = BinaryWriter::toValue(T(), AssumeVersion(g_network->protocolVersion())).size();
+	return value;
+}
+
+/**
+ * @class HeaderedItemsSerializerBase
+ *
+ * Encode objects in the format
+ *
+ *  | Header | Item | Item | ... |
+ * 
+ * The Header MUST have a fixed size when serialized.
+ */
+template <typename Header, typename Item>
+class HeaderedItemsSerializerBase {
+	BinaryWriter writer;
+
+	/// If true, no more items should be written to the writer
```

Comment:
This comment style is strange, why 3 '/' instead of 2?

Reply:
I am following Doxygen format, this is a shortcut of
```
/**
 * @brief
 */
```

Reply:
Probably should ask on community channel if we should follow certain documentation style, so that whatever we use stays consistent. Right now, I think we are somewhat close to https://google.github.io/styleguide/cppguide.html#Class_Comments than doxygen.

Reply:
I tend to agree. The current comment style is not doxygen and is more like google's C++ style.